core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -121,16 +121,18 @@ object BrokerToControllerChannelManager {
     config: KafkaConfig,
     channelName: String,
     threadNamePrefix: Option[String],
-    retryTimeoutMs: Long
-  ): BrokerToControllerChannelManager = {
+    retryTimeoutMs: Long,
+    networkClientRetryTimeoutMs: Option[Int] = None
```

Comment:
The fact that there are actually two separate timeouts here is a signal that something may be wrong.  To be clear, there were 2 separate timeouts prior to this PR -- this patch just makes their existence explicit so that we can set both of them in the KRaft case.   I've opened https://issues.apache.org/jira/browse/KAFKA-14394 to track this.

Reply:
looking at this again, I think we should just rename this to `requestTimeoutMs` and use it at the `NetworkManager` layer, as well as here.